### PR TITLE
Output is a URL so replace any backslash with slash

### LINKED ISSUE #1178: AnnotationProcessingCompilerToolTest fails on Windows since I20230621-1800
See
- https://download.eclipse.org/eclipse/downloads/drops4/I20230621-1800/testResults.php
- https://download.eclipse.org/eclipse/downloads/drops4/I20230621-1800/testresults/html/org.eclipse.jdt.apt.tests_ep429I-unit-win32-java17_win32.win32.x86_64_17.html

Looks line new line difference (Win/Unix) is the problem:
```
--------- Difference is ----------
expected:<...]\n
in jar:file:C:[\Users\genie.releng\AppData\Local\Temp\]org.eclipse.jdt.apt....> but was:<...]\n
in jar:file:C:[/Users/genie.releng/AppData/Local/Temp/]org.eclipse.jdt.apt....>
at org.eclipse.jdt.core.tests.junit.extension.TestCase.assertStringEquals(TestCase.java:266)
at org.eclipse.jdt.core.tests.junit.extension.TestCase.assertEquals(TestCase.java:242)
at org.eclipse.jdt.core.tests.compiler.regression.AbstractBatchCompilerTest.runTest(AbstractBatchCompilerTest.java:611)
at org.eclipse.jdt.apt.tests.AnnotationProcessingCompilerToolTest.runTest(AnnotationProcessingCompilerToolTest.java:212)
at org.eclipse.jdt.apt.tests.AnnotationProcessingCompilerToolTest.test_github844(AnnotationProcessingCompilerToolTest.java:63)
```

This is the new test added in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/845.

@guw : could you please take a look?

### PR DESCRIPTION
**Pull request by guw**
Fixes #1178

### CODE REVIEW

**Comment by iloveeclipse** [comment]
Hmm. I see on Windows following fail:

image

**Comment by guw** [comment]
@iloveeclipse I'll fix the first slash before the device. However, can you check if tests are executed differently in CI? I don't see the `\r` being reported as delta in the test failure report.

**Comment by iloveeclipse** [comment]
> I don't see the `\r` being reported as delta in the test failure report.

With CI do you mean Windows SDK build/test, or Jenkins? In Jenkins we only have Linux.

**Comment by iloveeclipse** [comment]
I've just started your test (actually entire test suite) from IDE and the only test complained about line endings was the new test.

**Comment by guw** [comment]
I mean the test report here:
https://download.eclipse.org/eclipse/downloads/drops4/I20230621-1800/testresults/html/org.eclipse.jdt.apt.tests_ep429I-unit-win32-java17_win32.win32.x86_64_17.html

```
--------- Difference is ----------
expected:<...]\n
in jar:file:C:[\Users\genie.releng\AppData\Local\Temp\]org.eclipse.jdt.apt....> but was:<...]\n
in jar:file:C:[/Users/genie.releng/AppData/Local/Temp/]org.eclipse.jdt.apt....>
```

It only complains about the slash vs. backslash but not the `\r`

File: org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/AnnotationProcessingCompilerToolTest.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public void test_github844() throws IOException {
 			+ "	last round: false\n"
 			+ "Discovered processor service org.eclipse.jdt.apt.tests.external.annotations.batch.BatchGenProcessor\n"
 			+ "  supporting [org.eclipse.jdt.apt.tests.external.annotations.batch.BatchGen]\n"
-			+ "  in jar:file:"  + _extJar.getCanonicalPath() +"!/\n"
+			+ "  in jar:"  + Path.of(_extJar.getCanonicalPath()).toUri().toURL().toString() +"!/\n"
```
Comment (guw) [inline org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/AnnotationProcessingCompilerToolTest.java:96]: @iloveeclipse Can you check if this produced the correct result on windows?

Comment (iloveeclipse) [reply org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/AnnotationProcessingCompilerToolTest.java:96]: Interestingly, after switching launch config to use windows-1252 from UTF-8 the test did not fail anymore.
??? Why character encoding affects new lines ???

Anyway, looks OK now.

Comment (guw) [reply org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/AnnotationProcessingCompilerToolTest.java:96]: Thanks for checking!

**Review by iloveeclipse — APPROVED** [review]

**Comment by iloveeclipse** [comment]
Thanks Gunnar.

**Pull request merged**